crates/ty_ide/src/goto.rs

```diff
@@ -947,29 +938,25 @@ pub(crate) fn find_goto_target(
     }
 
     let covering_node = covering_node(parsed.syntax().into(), token.range())
-        .find_first(|node| node.is_identifier() || node.is_expression())
+        .find_first(|node| {
+            node.is_identifier() || node.is_expression() || node.is_stmt_import_from()
```

Comment:
I wonder if it makes sense to change this to `node.is_stmt`. I don't think we ever want to go beyond the closest statement (my, now proven incorrect, assumption was that `expression` is a sufficient boundary but turns out it isn't)